packages/connect-node-test/src/crosstest/cacheable_unary.spec.ts

```diff
@@ -0,0 +1,113 @@
+      "@bufbuild/connect-node (gRPC-web, binary, http) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC-web, binary, http2, gzip) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC-web, binary, http2) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC-web, binary, https) against @bufbuild/connect-node (h1 + tls)",
+      "@bufbuild/connect-node (gRPC-web, JSON, http, gzip) against @bufbuild/connect-express (h1)",
+      "@bufbuild/connect-node (gRPC-web, JSON, http, gzip) against @bufbuild/connect-fastify (h2c)",
+      "@bufbuild/connect-node (gRPC-web, JSON, http, gzip) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC-web, JSON, http) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC-web, JSON, http2, gzip) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC-web, JSON, http2) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC-web, JSON, https) against @bufbuild/connect-node (h1 + tls)",
+      "@bufbuild/connect-node (gRPC, binary, http, gzip) against @bufbuild/connect-express (h1)",
+      "@bufbuild/connect-node (gRPC, binary, http, gzip) against @bufbuild/connect-fastify (h2c)",
+      "@bufbuild/connect-node (gRPC, binary, http, gzip) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC, binary, http) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC, binary, http2, gzip) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC, binary, http2) against @bufbuild/connect-node (h2)",
+      "@bufbuild/connect-node (gRPC, binary, http2) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC, binary, https) against @bufbuild/connect-node (h1 + tls)",
+      "@bufbuild/connect-node (gRPC, JSON, http, gzip) against @bufbuild/connect-express (h1)",
+      "@bufbuild/connect-node (gRPC, JSON, http, gzip) against @bufbuild/connect-fastify (h2c)",
+      "@bufbuild/connect-node (gRPC, JSON, http, gzip) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC, JSON, http) against @bufbuild/connect-node (h1)",
+      "@bufbuild/connect-node (gRPC, JSON, http2, gzip) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC, JSON, http2) against @bufbuild/connect-node (h2c)",
+      "@bufbuild/connect-node (gRPC, JSON, https) against @bufbuild/connect-node (h1 + tls)",
+    ],
+    (transportFactory) => {
+      const request = new SimpleRequest({
+        responseSize: 1024,
+        payload: {
+          body: new Uint8Array(1024).fill(0),
+        },
+      });
+      it("with promise client", async function () {
+        const transport = transportFactory({
+          useHttpGet: true,
+          interceptors: [ensureGetRequest],
+        });
+        const client = createPromiseClient(TestService, transport);
```

Comment:
The interceptor is a valid and nice solution, but the preferred way to access response headers is via a call option, and I think we should prefer it in the tests too. 

For example:

```ts
function ensureGetRequest(header: Headers) {
  expect(header.has("request-protocol")).toBeTrue();
  if (header.get("request-protocol") === "connect") {
    expect(header.has("get-request")).toBeTrue();
  }
}
```

With:

```ts
const response = await client.cacheableUnaryCall(request, {onHeader: ensureGetRequest});
```

Could you update the four places?

Reply:
Thanks, that is a great improvement. Done.